eclipse-che4z/che-che4z-lsp-for-cobol
# feat: Create command to run analysis vscode command via CLI/terminal

Adds a new command to the command palette such that the user can easily trigger the analysis from the CLI. Handles both the Java and Native versions of the server along with saved and unsaved files.

## How Has This Been Tested?

Functionality:
1. Open VS code with a cobol file open as the active file/editor
2. Press F1 to open the command palette
3. Type "cobol analysis" and select that command
4. Select either "Java" or "Native"
5. Press enter, see the output in the terminal window at the bottom. (Default UI location)

Tests created:

- [x] Cobol Analysis - Java: Tests when the user would have selected "Java" as the environment
- [x] Cobol Analysis - Native: Tests when the user would have selected "Native" as the environment
- [x] Cobol Analysis - Undefined Type: Tests when the user would have backed out of the command without selecting anything
- [x] Cobol Analysis - Save temp file: Tests what happens when the user tries to run the command on an unsaved file.
- [x] Various functions to test/determine functionality.

## Checklist:
